client/src/services/categories.service.ts

```diff
@@ -4,7 +4,7 @@ import { Category } from "../models/category";
 
 export const getCategories = (gender : Gender | undefined) => {
     return axiosInstance.get(`/categories`).then((response) => {
-        return response.data.map((category : Category) => {
+        return response.data?.map((category : Category) => {
```

Comment:
Why?

Reply:
avoid error if the server not return categories